### ISSUE TITLE
[Bug]: ArticleReaction.vue:99 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'reaction0')

### ISSUE THREAD

**Issue by wish5115** [walinejs/waline#2466]
### 问题描述 | Describe the bug

**The following error occurs when loading a page or switching the document in the left sidebar:**
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'reaction0')
    at ArticleReaction.vue:99:33
    at Array.map (<anonymous>)
    at f (ArticleReaction.vue:98:32)
(anonymous) @ ArticleReaction.vue:99
f @ ArticleReaction.vue:98
await in f (async)
Pe.immediate @ ArticleReaction.vue:152
ut @ runtime-core.esm-bundler.js:192
Be @ runtime-core.esm-bundler.js:199
w @ runtime-core.esm-bundler.js:1941
Hs @ runtime-core.esm-bundler.js:1979
Pe @ runtime-core.esm-bundler.js:1806
(anonymous) @ ArticleReaction.vue:149
(anonymous) @ runtime-core.esm-bundler.js:2872
ut @ runtime-core.esm-bundler.js:192
Be @ runtime-core.esm-bundler.js:199
Ka.t.__weh.t.__weh @ runtime-core.esm-bundler.js:2852
Os @ runtime-core.esm-bundler.js:367
qt @ runtime-core.esm-bundler.js:6676
mount @ runtime-core.esm-bundler.js:3928
t.mount @ runtime-dom.esm-bundler.js:1501
Od @ init.ts:81
window.initWaline @ (index):18
(anonymous) @ (index):18
Show 13 more frames
Show less
```

**This error, refresh the page may not be easy to appear, switch the left side of the document is easy to appear, or switch the document and then refresh the page after the switch is also easy to appear, it is recommended to try a few more times, you will be able to find the error.**

For example:

https://wxkj.asia/mynote/%E5%BC%80%E5%8F%91%E7%AC%94%E8%AE%B0/%E5%89%8D%E7%AB%AF/%E9%98%B2%E6%AD%A2window.onload%E8%A2%AB%E8%A6%86%E7%9B%96%E7%9A%84%E6%8A%80%E5%B7%A7.html

You can click on the URL above and refer to my GIF image:

![waline-bugs](https://github.com/walinejs/waline/assets/5248206/d89d381b-fdf4-4f22-88cd-2d55280671c1)

Here resp is an object, not an array, so [0] here is redundant, I think.

<img width="665" alt="Snipaste_2024-05-02_17-07-27" src="https://github.com/walinejs/waline/assets/5248206/c9601796-0db6-4493-9e4c-1ad8b064da49">

This bug is the same as the previous 2451 bug, but 2451 was closed, so it was rebuilt. see: [2451](https://github.com/walinejs/waline/issues/2451)

### 问题网站 | Website URL

https://wxkj.asia/mynote/%E5%BC%80%E5%8F%91%E7%AC%94%E8%AE%B0/%E5%89%8D%E7%AB%AF/%E9%98%B2%E6%AD%A2window.onload%E8%A2%AB%E8%A6%86%E7%9B%96%E7%9A%84%E6%8A%80%E5%B7%A7.html

### 服务部署在哪里？ | Where your waline deploy?

Vercel (Default)

### 数据存储在哪里？| Where your comment data store?

LeanCloud(https://leancloud.app)

**Comment by track23** [comment]
有没有可能这是你自己的bug。你不是已经知道原因了吗。resp不是数组

**Comment by wish5115** [comment]
> 有没有可能这是你自己的bug。你不是已经知道原因了吗。resp不是数组
> 
> Is it possible that this is your own bug. Don't you already know the reason? resp is not an array

<img width="738" alt="image" src="https://github.com/walinejs/waline/assets/5248206/bb007023-2510-43a5-aeb0-1b0ae4f92286">

你看这个文件路径，说明不是我的文件。而且，如果是我的文件，我自己就改掉了，我还会到这里提bug吗？

---

If you look at this file path, it means it's not my file. Besides, if it was my file, I would have changed it myself, would I have come here to mention the bug?

**Comment by lizheming** [comment]
@track23 这个同学之前提过一个 issue 我这边跟进了下，后续确认是 Waline 初始数据的问题，影响不大，只有在没有 reaction 数据的时候会有这个报错，对用户使用没有影响。这个 issue 先放着，后面我空了处理下。感谢你帮忙跟进~

**Comment by observerkei** [comment]
这个问题解决了吗？ "@waline/client": "^3.4.2" 里面也碰到了这个问题。

---  

Has the problem been solved? "@waline/client": "^3.4.2" also runs into this problem.

**Comment by lizheming** [comment]
@observerkei 需要升级 `@waline/vercel`
